pilot/pkg/xds/debuggen.go

```diff
@@ -0,0 +1,122 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package xds
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	"github.com/golang/protobuf/ptypes/any"
+
+	"istio.io/istio/pilot/pkg/model"
+)
+
+const (
+	// TypeDebug requests debug info from istio, a secured implementation for istio debug interface
+	TypeDebug = "istio.io/debug"
+	debugURL  = "http://localhost:8080/debug/"
+)
+
+var activeNamespaceDebuggers = []string{
```

Comment:
can you make this a `map[string]struct{}` so lookup doesn't need a loop?

Reply:
fixed